akka-http-caching/src/main/scala/akka/http/caching/javadsl/CachingSettings.scala

```diff
@@ -35,8 +37,10 @@ abstract class LfuCacheSettings { self: LfuCacheSettingsImpl ⇒
   def withTimeToIdle(newTimeToIdle: Duration): LfuCacheSettings
 }
 
-object CachingSettings {
-  /** Java API: Create default caching settings from the systems' configuration */
-  def create(system: ActorSystem): CachingSettings =
-    akka.http.caching.scaladsl.CachingSettings(system)
+object CachingSettings extends SettingsCompanion[CachingSettings] {
+  override def create(config: Config): CachingSettings =
+    akka.http.caching.scaladsl.CachingSettings(config)
+  override def create(configOverrides: String): CachingSettings =
+    akka.http.caching.scaladsl.CachingSettings(configOverrides)
+  override def create(system: ActorSystem): CachingSettings = create(system.settings.config)
 }
```

Comment:
nice, thanks; adding the Java API comment would be nice too

Reply:
True, I thought genjavadoc would somehow put them from the overridden methods, but it seems not.